code/modules/admin/battle_royale.dm

```diff
@@ -0,0 +1,429 @@
+	GLOB.battle_royale = new()
+	GLOB.battle_royale.start()
+
+/client/proc/battle_royale_speed()
+	set name = "Battle Royale - Change wall speed"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	var/new_speed = input(src, "New wall delay (seconds)") as num
+	if(new_speed > 0)
+		GLOB.battle_royale.field_delay = new_speed
+		log_admin("[key_name(usr)] has changed the field delay to [new_speed] seconds")
+		message_admins("[key_name(usr)] has changed the field delay to [new_speed] seconds")
+
+/client/proc/battle_royale_varedit()
+	set name = "Battle Royale - Variable Edit"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	debug_variables(GLOB.battle_royale)
+
+/client/proc/battle_royale_spawn_loot()
+	set name = "Battle Royale - Spawn Loot Drop (Minor)"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	GLOB.battle_royale.generate_good_drop()
+	log_admin("[key_name(usr)] generated a battle royale drop.")
+	message_admins("[key_name(usr)] generated a battle royale drop.")
+
+/client/proc/battle_royale_spawn_loot_good()
+	set name = "Battle Royale - Spawn Loot Drop (Major)"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	GLOB.battle_royale.generate_endgame_drop()
+	log_admin("[key_name(usr)] generated a good battle royale drop.")
+	message_admins("[key_name(usr)] generated a good battle royale drop.")
+
+/datum/battle_royale_controller
+	var/list/players
+	var/datum/proximity_monitor/advanced/battle_royale/field_wall
+	var/radius = 118
+	var/process_num = 0
+	var/list/death_wall
+	var/field_delay = 15
+	var/debug_mode = FALSE
+
+/datum/battle_royale_controller/Destroy(force, ...)
+	QDEL_LIST(death_wall)
+	for(var/client/C in GLOB.admins)
+		C.verbs -= BATTLE_ROYALE_AVERBS
+	. = ..()
+	GLOB.enter_allowed = TRUE
+	world.update_status()
+	GLOB.battle_royale = null
+
+//Trigger random events and shit, update the world border
+/datum/battle_royale_controller/process()
+	process_num ++
+	//Once every 50 seconds
+	if(prob(2))
+		generate_basic_loot(2)
+	//Once every 300 seconds
+	if(prob(0.333))
+		generate_good_drop()
+	var/living_victims = 0
+	var/winner
+	for(var/mob/living/M as anything in players)
```

Comment:
```suggestion
	for(var/mob/living/M as() in players)
```
Apparently there's a BYOND bug with `as anything` in some cases, so it's better to get in the habit of using `as()`

Reply:
I thought you said it was the other way round